### ISSUE TITLE
Chain middleware does not work and replies with 404

### ISSUE THREAD

**Issue by tomfun** [containous/traefik#5280]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

I create ingress route and 2 middlewares, route work well, basic auth middleware work well
```yml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: meow-svc-https-wc-mtls
spec:
  entryPoints:
    - https
  routes:
    - match: Host(`mtls-meow.EXAMPLE.com`)
      kind: Rule
      priority: -1
      services:
        - name: meow-svc
          port: 80
      middlewares:
        - name: global
## But this line work well (without previous line):
#        - name: test-middleware
  tls: {}

---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: global
spec:
  chain:
    middlewares:
    - "test-middleware"

---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: test-middleware
spec:
  basicAuth:
    users:
      - test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/

```
So I can turn off chain middleware, and it is asks me enter login pass, or gives 401.

### What did you expect to see?

I expect request for login password or 401 status.

### What did you see instead?

404 status

### Output of `traefik version`: 2.0 RC1

```
# kubectl exec tr-ingrs-realease-traefik-74d87f59cc-9jgll /traefik version
Version:      2.0.0-rc1
Codename:     montdor
Go version:   go1.13rc1
Built:        2019-08-26T17:47:29Z
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml

    [Global]
        checkNewVersion = true
        sendAnonymousUsage = false

    [log]
      level = "debug"
      format = "json"

    [entryPoints]
        [entryPoints.http]
          address = ":80"
        [entryPoints.https]
          address = ":443"

    [providers]
        [providers.kubernetesIngress]
        [providers.kubernetesCRD]

    [accessLog]

    [ping]

    [http]
        [http.routers]
           [http.routers.my-router]
              rule = "Path(`/test`)"
              service = "whoami"
              entrypoint=["http"]

        [http.services]
              [http.services.whoami.loadbalancer]
                [[http.services.whoami.loadbalancer.servers]]
                  url = "http://localhost:8080"

    [certificatesResolvers]
      [certificatesResolvers.defaultResolver]
        [certificatesResolvers.defaultResolver.acme]
            keyType = "RSA4096"
            email = "tomfun1990@gmail.com"
            storage = "/acme/acme_2.0.json"
            [certificatesResolvers.defaultResolver.acme.dnsChallenge]
                provider = "route53"
            [certificatesResolvers.defaultResolver.acme.tlsChallenge]

    [api]
      dashboard = true

    [tls.options]
      [tls.options.default]
          minVersion = "VersionTLS12"
          cipherSuites = [
            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
            "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
            "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
            "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256"
          ]
          CertFile = "/ssl/tls.crt"
          KeyFile = "/ssl/tls.key"
      [tls.options.mtls]
          minVersion = "VersionTLS12"
          cipherSuites = [
            "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
            "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
            "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
            "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
            "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
            "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
            "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256"
          ]
          CertFile = "/ssl/tls.crt"
          KeyFile = "/ssl/tls.key"
          [tls.options.mtls.clientAuth]
            caFiles = ["/mtls/clientCaCert-0.crt"]
            clientAuthType = "RequireAndVerifyClientCert"
```
### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)
```
{"level":"debug","msg":"No secret name provided","providerName":"kubernetescrd","time":"2019-09-02T11:37:54Z"}
{"level":"debug","msg":"No secret name provided","providerName":"kubernetescrd","time":"2019-09-02T11:37:54Z"}
{"level":"debug","msg":"Skipping Kubernetes event kind *v1.Endpoints","providerName":"kubernetescrd","time":"2019-09-02T11:37:54Z"}
{"level":"debug","msg":"Skipping Kubernetes event kind *v1.Endpoints","providerName":"kubernetes","time":"2019-09-02T11:37:54Z"}
192.168.157.148 - - [02/Sep/2019:11:37:54 +0000] "GET / HTTP/2.0" 404 19 "-" "-" 2262 - - 0ms
```

**Comment by dduportal** [comment]
Hi @tomfun , thanks for your feedback. So the issue **was** that the object `middleware` should have been the following, assuming that your default Kubernetes namespace is `default` (note that the middleware reference is prefixed with the namespace:

```yml
----
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: global
spec:
  chain:
    middlewares:
    - "default/test-middleware"
```

As this could be a cumbersome user experience, the team sent the pull request #5290 which changes the syntax to avoid such "implicit" error, that will fix this issue: this why we've labelled it both as a question (user question where documentation was not precise enough) AND an enhancement for next v2.0-rc.

**Comment by traefiker** [comment]
Closed by #5290.
